operator/pkg/resources/statefulset.go

```diff
@@ -502,6 +502,8 @@ func (r *StatefulSetResource) obj(
 						{
 							Name:            configuratorContainerName,
 							Image:           r.fullConfiguratorImage(),
+							Command:         []string{"/redpanda-operator"},
+							Args:            []string{"configure"},
```

Comment:
So, since we're modifying the entrypoint of the configurator -- I believe (assuming we also use it in v2) we'll need to mirror this change in the Redpanda helm chart as well (do we want to do that before or after the rsync work in the helm-charts repo?).

That said, I think this warrants a callout in a changelog, specifically stating that if you're referencing a custom configurator image you'll need to:

1. Leverage the additionalCmdFlags to override the configurator image as the other flag is now deprecated
2. Ensure that the mirrored container supports this entrypoint (maybe just say that the configurator image _must_ be based on vX.Y.Z of the operator image?)

Reply:
> I believe (assuming we also use it in v2) we'll need to mirror this change in the Redpanda helm chart as well

We don't use this exact configurator between v1 and v2. 

v1 uses https://github.com/redpanda-data/redpanda-operator/tree/2b2277399bdb6069608e1a3e1af2b6cb8864ed70/operator/cmd/configurator

v2 uses https://github.com/redpanda-data/redpanda-operator/tree/2b2277399bdb6069608e1a3e1af2b6cb8864ed70/operator/cmd/envsubst

> That said, I think this warrants a callout in a changelog

I will add change log statement about configurator entrypoint.